src/app/compose/draftdesk.service.spec.ts

```diff
@@ -33,56 +33,209 @@ describe('DraftDesk', () => {
+                ]
+                ,
+                to: [
+                    {address: 'to@runbox.com', name: 'To'}
+                ],
+                date: mailDate,
+                subject: 'Test subject',
+                text: 'blabla\nabcde',
+                rawtext: 'blabla\nabcde',
+                html: '<p>blabla</p><p>abcde</p>'
+            }
+            , [ FromAddress.fromEmailAddress('to@runbox.com')]
+        , false, false);
+
+        expect(draft.subject).toBe('Re: Test subject');
+        expect(draft.from).toBe('to@runbox.com');
+        expect(draft.to[0].nameAndAddress).toBe('"Reply-To" <reply-to@runbox.com>');
+        expect(draft.msg_body).toBe(`\n2017-07-01 00:00 ${timezoneOffsetString} "From" <from@runbox.com>:\n> blabla\n> abcde`);
+        expect(draft.isUnsaved()).toBe(true);
+        done();
+    });
+    it('Reply: Address with object, reply to all', (done) => {
+        console.log('Reply test: Address with object, reply to all');
+        const mailDate = new Date(2017, 6, 1);
+
+        const timezoneOffset: number = mailDate.getTimezoneOffset();
+
+        const timezoneOffsetString: string = 'GMT' + (timezoneOffset <= 0 ? '+' : '-') +
+            ('' + (100 + (Math.abs(timezoneOffset) / 60))).substr(1, 2) + ':' +
+            ('' + (100 + (Math.abs(timezoneOffset) % 60))).substr(1, 2);
+
+        const draft = DraftFormModel.reply({
+                headers: {
+                    'message-id': 'themessageid12123abcdef',
+                },
+                from: [
+                    {address: 'from@runbox.com', name: 'From'}
+                ]
+                ,
+                to: [
+                    {address: 'to@runbox.com', name: 'To'}
+                ],
+                date: mailDate,
+                subject: 'Test subject',
+                text: 'blabla\nabcde',
+                rawtext: 'blabla\nabcde',
+                html: '<p>blabla</p><p>abcde</p>'
+            }
+            , [ FromAddress.fromEmailAddress('to@runbox.com')]
         , true, false);
 
         expect(draft.subject).toBe('Re: Test subject');
-        expect(draft.to).toBe('Test1<test1@runbox.com>');
-        expect(draft.msg_body).toBe(`\n2017-07-01 00:00 ${timezoneOffsetString} Test1<test1@runbox.com>:\n> blabla\n> abcde`);
+        expect(draft.from).toBe('to@runbox.com');
+        expect(draft.to[0].nameAndAddress).toBe('"From" <from@runbox.com>');
+        expect(draft.msg_body).toBe(`\n2017-07-01 00:00 ${timezoneOffsetString} "From" <from@runbox.com>:\n> blabla\n> abcde`);
         expect(draft.isUnsaved()).toBe(true);
-        draft = DraftFormModel.reply({
+        done();
+    });
+    it('Reply: Address with MAI', (done) => {
+        console.log('Reply test: Address with MAI');
+        const mailDate = new Date(2017, 6, 1);
+
+        const timezoneOffset: number = mailDate.getTimezoneOffset();
+
+        const timezoneOffsetString: string = 'GMT' + (timezoneOffset <= 0 ? '+' : '-') +
```

Comment:
This piece is duplicated in this file, could probably be refactored into a function.

I wonder, though, if we really need all this computing? Do the tests care for something more than "the timezone is the same before and after?"

Reply:
Good point, I just duplicated what was already there..